include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -160,13 +167,48 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+def GetParentOp : IbisOp<"get_parent"> {
+  let summary = "Ibis get parent";
+  let description = [{
+    Given an Ibis class reference, returns the parent of said class. The
+    returned `ibis.classref` will be an opaque type.
+  }];
+
+  let arguments = (ins AnyClassRefType:$src);
+  let results = (outs OpaqueClassRefType:$parent);
+
+  let assemblyFormat = [{
+    $src `:` qualified(type($src)) attr-dict
+  }];
+}
+
+def GetChildOp : IbisOp<"get_child"> {
```

Comment:
"child" is somewhat ambiguous in this case. May use the term "instance" which is more precise.